src/Microsoft.CodeQuality.Analyzers/UnitTests/QualityGuidelines/AvoidDuplicateElementInitializationTests.cs

```diff
@@ -0,0 +1,282 @@
+        {
+            [System.DateTimeKind.Local] = 1,
+            [System.DateTimeKind.Utc] = 2,
+            [System.DateTimeKind.Local] = 3
+        };
+    }
+}
+",
+                // TODO: See if there's a way to use 'DateTimeKind.Local' here.
+                GetCSharpResultAt(10, 13, "2"));
+        }
+
+        [Fact]
+        public void MultipleIndexerArguments()
+        {
+            VerifyCSharp(@"
+class C
+{
+    void Foo()
+    {
+        var x = new D
+        {
+            [1, ""a""] = 1,
+            [1, ""b""] = 2,
+            [2, ""a""] = 3,
+            [1, ""a""] = 4
+        };
+    }
+}
+
+class D
+{
+    public int this[int a, string b]
+    {
+        get { return 0; }
+        set { }
+    }
+}
+",
+                GetCSharpResultAt(11, 13, "1, a"));
+        }
+
+        [Fact]
+        public void MultipleIndexerArgumentsNamed()
+        {
+            VerifyCSharp(@"
+class C
+{
+    void Foo()
+    {
+        var x = new D
+        {
+            [""a"", ""b""] = 1,
+            [""b"", ""c""] = 2,
+            [b: ""a"", a: ""b""] = 3,
+            [b: ""b"", a: ""a""] = 4
+        };
+    }
+}
+
+class D
+{
+    public int this[string a, string b]
+    {
+        get { return 0; }
+        set { }
+    }
+}
+",
+                GetCSharpResultAt(11, 13, "a, b"));
+        }
+
+        [Fact]
+        public void MultipleIndexerArgumentsNamedWithAtPrefix()
+        {
+            VerifyCSharp(@"
+class C
+{
+    void Foo()
+    {
+        var x = new D
+        {
+            [""a"", ""b""] = 1,
+            [""b"", ""c""] = 2,
+            [@class: ""a"", a: ""b""] = 3,
+            [@class: ""b"", @a: ""a""] = 4
+        };
+    }
+}
+
+class D
+{
+    public int this[string a, string @class]
+    {
+        get { return 0; }
+        set { }
+    }
+}
+",
+                GetCSharpResultAt(11, 13, "a, b"));
+        }
+
+        [Fact]
+        public void MultipleArgumentsWithOmittedDefault()
+        {
+            VerifyCSharp(@"
+class C
+{
+    void Foo()
+    {
+        var x = new D
+        {
+            [""a"", ""b""] = 1,
+            [""b"", ""c""] = 2,
+            [b: ""a"", a: ""b""] = 3,
+            [""a""] = 4
+        };
+    }
+}
+
+class D
+{
+    public int this[string a, string b = ""b""]
+    {
+        get { return 0; }
+        set { }
+    }
+}
+",
+                GetCSharpResultAt(11, 13, "a, b"));
+        }
+
+        [Fact]
+        public void NonConstantArguments()
+        {
+            VerifyCSharp(@"
+class C
+{
+    void Foo()
+    {
+        var dictionary = new System.Collections.Generic.Dictionary<string, int>
+        {
+            [""a""] = 1,
+            [typeof(C).ToString()] = 2,
+            [""a""] = 3
+        };
+    }
+}
+",
+                GetCSharpResultAt(10, 13, "a"));
+        }
+
+        private DiagnosticResult GetCSharpResultAt(int line, int column, string symbolName)
```

Comment:
Can you please add VB tests or open a tracking issue to add them later?

Reply:
I still don't know how to do this in VB. I asked above and you pinged @AlekseyTs for guidance [here](https://github.com/dotnet/roslyn-analyzers/pull/1142#issuecomment-293981487), but we have not received it yet.

Reply:
In that case, can you change the DiagnosticAnalyzer attribute to not specify VB and open a separate tracking item to figure out VB story/tests?

Reply:
Done